src/Config.cpp

```diff
@@ -0,0 +1,141 @@
+      return Nil{};
+    }
+
+    auto
+    parse(FileConfig &conf, const std::filesystem::path &path) -> MaybeError
+    {
+      auto json = TRY(JSON::Parser::load_from_file(path));
+      auto ingredients_array = TRY(json.get<JSON::JSONArray>("ingredients"));
+      auto recipes_object = TRY(json.get<JSON::JSONObject>("recipes"));
+
+      for (const auto &item: TRY(json.get<JSON::JSONObject>()))
+        if (item.first != "ingredients" && item.first != "recipes")
+          return Error("Invalid key in JSON");
+
+      if (ingredients_array.empty())
+        return Error("Ingredients array is empty");
+      if (recipes_object.empty())
+        return Error("Recipes object is empty");
+
+      conf.ingredientsIds = std::
+        make_unique<Data::Ids>(ingredients_array.size());
+      conf.recipesIds = std::make_unique<Data::Ids>(recipes_object.size());
+
+      for (const auto &item: ingredients_array) {
+        const auto ingredient = TRY(item.get<std::string>());
+        auto id = conf.ingredientsIds->lookup(ingredient);
+        if (id.has_value())
+          return Error("Duplicate ingredient found");
+        conf.ingredientsIds->add(ingredient);
+      }
+
+      for (const auto &[key, value]: recipes_object)
+        TRY(parseRecipe(conf, key, value));
+
+      return Nil{};
+    }
+  }  // namespace
+
+  auto Init::fileConf() -> ErrorOr<FileConfig>
+  {
+    FileConfig conf;
+
+    auto conf_parsing = parse(conf, "config.json");
+    if (conf_parsing.is_error())
+      MUST(parse(conf, "default.json"), "Default.json not found or invalid");
+    return conf;
+  }
+
+  namespace {
+
+#define IS_ONLY(str, valid) (strspn(str, valid) == strlen(str))
+
+    auto validateIntArgs(char *arg) -> MaybeError
+    {
+      if (!IS_ONLY(arg, "0123456789"))
+        return Error("Time parameter is not a positive intger");
+      if (std::stoul(arg, nullptr, 10) == 0)
+        return Error("Time parameter cannot be 0");
+      return Nil{};
+    }
+
+    auto validateDoubleArgs(char *arg) -> MaybeError
+    {
+      size_t idx = 0;
+      double value = 0;
+
+      try {
+        value = std::stod(arg, &idx);
+      } catch (...) {
+        return Error("Multiplier parameter is not a double");
+      }
+      if (idx != strlen(arg))
+        return Error("Multiplier parameter is not a double");
+      if (value <= 0)
+        return Error("Multiplier parameter cannot be 0 or negative");
+      return Nil{};
+    }
+
+#undef IS_ONLY
+
+  }  // namespace
+
+  auto Init::argConf(int argc, std::span<char *> argv) -> ErrorOr<Params>
+  {
+    Config::Params params;
+    if (argc != 4)
+      return Error("Not enough arguments");
+
+    TRY(validateDoubleArgs(argv[1]));
+    TRY(validateIntArgs(argv[2]));
+    TRY(validateIntArgs(argv[3]));
+
+    params.multiplier = std::stod(argv[1], nullptr);
+    params.cook = std::stoul(argv[2], nullptr, 10);
+    params.time = std::chrono::milliseconds(std::stoul(argv[3], nullptr, 10));
```

Comment:
```suggestion
    params.multiplier = TRY(validateDoubleArgs(argv[1]));
    params.cook = TRY(validateIntArgs(argv[2]));
    params.time = std::chrono::milliseconds(TRY(validateIntArgs(argv[3])));
```
Adapt that to use the return value!